app/workers/pivotal_worker.rb

```diff
@@ -1,47 +1,15 @@
 class PivotalWorker
-include Sidekiq::Worker
-sidekiq_options retry: 1
+  include Sidekiq::Worker
+  sidekiq_options retry: 1
 
   def perform(user_id)
     user = User.find(user_id)
 
-    @client = TrackerApi::Client.new(token: user.pivotal_token)
+    pivotal_service = PivotalApiService.new(user.pivotal_token)
 
     @first_run = user.pivotal_id.nil?
-    user.update(pivotal_id: @client.me.id) unless user.pivotal_id
+    user.update(pivotal_id: pivotal_service.user_pivotal_id) if @first_run
```

Comment:
Don't we want to have the same logic here as before? with the new logic, we will run the update if `user.pivotal_id` is `nil` ?